File: python/secret-handshake/secret_handshake.py
def commands(binary_str):
    value = int(binary_str, 2)
    responses = ['wink', 'double blink', 'close your eyes', 'jump']
    actions = [responses[i] for i in range(4) if 1 << i & value]
    return actions[::-1] if 1 << 4 & value else actions

File: ruby/secret-handshake/secret_handshake.rb
class SecretHandshake
  def initialize(input)
    @num = input.is_a?(Integer) ? input % 32 : 0
  end

  def commands
    responses = ['wink', 'double blink', 'close your eyes', 'jump']
    actions = []
    (0..3).each { |i| actions.push(responses[i]) if @num & (1 << i) != 0 }
    1 << 4 & @num == 0 ? actions : actions.reverse
  end
end
